### ISSUE TITLE
Pretty printing HTML

### ISSUE THREAD

**Issue by RezwanArefin01** [yawaramin/dream-html#34]
Currently the produced HTML doesn't look good: it inserts a newline before each attribute. It would be nice to have a pretty printing option.

I am willing to work on this in a PR. My current plan is to change the `to_string` signature to:
```ocaml
val to_string : xml:bool -> indent_with:[< `Tabs | `Spaces of int ] -> node -> string
```
The algorithm would be roughly this:
- If current tag has no children: open and close the tag (appropriately), without a newline.
- If current tag's children has at least one `Txt` node: do not indent the children.
- Otherwise, indent the children.

The second rule is like that for cases like: `<p><b>Hello,</b> World</p>` should be pretty printed to the same thing. However, `<p><b>Hello, World</b></p>` should be pretty printed to:
```
<p>
  <b>Hello, World</b>
</p>
```
Basically, if all the children are tags or comments, we can safely indent them.

Let me know your inputs.

**Comment by yawaramin** [comment]
The algorithm seems fine, but instead of changing the function signature (breaking compatibility), we can just default to 2 spaces indent. Overall pretty-printing the HTML is a good idea and I agree we should do it.

**Comment by RezwanArefin01** [comment]
> instead of changing the function signature (breaking compatibility), we can just default to 2 spaces indent

Yes. I meant the internal `to_string` method that's there currently, which is later exported publicly with defaults. :slightly_smiling_face:

**Comment by RezwanArefin01** [comment]
Another issue is: the `Dream_html` module  has some functions (`write`, `set_body`, `respond`, `send`) that use the `to_string` method from `Pure_html`.  What would be preferable:
- All these functions accept the `~indent_with` argument.
- Or, have the `indent_with` as a global variable in `Pure_html` that is set with a function?

I prefer the second option. Because it is unlikely that someone needs different pretty-printing options in a project, while the other functions are likely to be used many times throughout the project.

**Comment by yawaramin** [comment]
There's no need to have an indent setting option, we would just default to 2-space indent everywhere :-)

**Comment by yawaramin** [comment]
Btw, if you prefer 4-space indent or tab indent, we can default to that, I'm not picky about that :-)
